pkg/lib/bundle/validate.go

```diff
@@ -132,11 +162,155 @@ func (i imageValidator) ValidateBundle(directory string) error {
 
 	_, err = ValidateChannelDefault(annotations[ChannelsLabel], annotations[ChannelDefaultLabel])
 	if err != nil {
-		annotationErrors = append(annotationErrors, err)
+		validationErrors = append(validationErrors, err)
+	}
+
+	if len(validationErrors) > 0 {
+		return NewValidationError(validationErrors)
+	}
+
+	return nil
+}
+
+// ValidateBundleContent confirms that the CSV and CRD files inside the bundle
+// directory are valid and can be installed in a cluster. Other GVK types are
+// also validated to confirm if they are "kubectl-able" to a cluster meaning
+// if they can be applied to a cluster using `kubectl` provided users have all
+// necessary permissions and configurations.
+// Inputs:
+// manifestDir: the directory which all bundle manifests files are located
+// Outputs:
+// error: ValidattionError which contains a list of errors
+func (i imageValidator) ValidateBundleContent(manifestDir string) error {
+	var validationErrors []error
+
+	i.logger.Debug("Validating bundle contents")
+
+	mediaType, err := GetMediaType(manifestDir)
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	switch mediaType {
+	case HelmType:
+		return nil
+	}
+
+	var csvName string
+	unstObjs := []*unstructured.Unstructured{}
+	csvValidator := v.ClusterServiceVersionValidator
+	crdValidator := v.CustomResourceDefinitionValidator
+
+	// Read all files in manifests directory
+	items, err := ioutil.ReadDir(manifestDir)
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	for _, item := range items {
+		fileWithPath := filepath.Join(manifestDir, item.Name())
+		data, err := ioutil.ReadFile(fileWithPath)
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("Unable to read file %s in supported types", fileWithPath))
+			continue
+		}
+
+		dec := k8syaml.NewYAMLOrJSONDecoder(strings.NewReader(string(data)), 30)
+		k8sFile := &unstructured.Unstructured{}
+		err = dec.Decode(k8sFile)
+		if err != nil {
+			validationErrors = append(validationErrors, err)
+			continue
+		}
+
+		unstObjs = append(unstObjs, k8sFile)
+		gvk := k8sFile.GetObjectKind().GroupVersionKind()
+		i.logger.Debugf(`Validating "%s" from file "%s"`, gvk.String(), item.Name())
+		// Verify if the object kind is supported for RegistryV1 format
+		ok, _ := IsSupported(gvk.Kind)
+		if mediaType == RegistryV1Type && !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("%s is not supported type for registryV1 bundle: %s", gvk.Kind, fileWithPath))
+			continue
+		}
+
+		if gvk.Kind == CSVKind {
+			csv := &v1.ClusterServiceVersion{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(k8sFile.Object, csv)
+			if err != nil {
+				validationErrors = append(validationErrors, err)
+			}
```

Comment:
A `continue` here will avoid duplicating errors from the validator below, since the unstructured converter will not populate `csv` if an error occurs.
```suggestion
				continue
			}
```

Reply:
Ack

Reply:
Resolved.